lib/semantics/tools.h

```diff
@@ -469,5 +469,43 @@ FindPolymorphicAllocatableUltimateComponent(const DerivedTypeSpec &);
 UltimateComponentIterator::const_iterator
 FindPolymorphicAllocatableNonCoarrayUltimateComponent(const DerivedTypeSpec &);
 
+class LabelEnforce {
```

Comment:
If this class will serve for the implementation of additional semantic checks, please add a line or two of commentary here describing what it does.  But if it's probably only going to be useful for `CRITICAL`, then it should not be in tools.h, since it uses the parse tree walking framework and that's somewhat expensive (in term of build time) to instantiate in all clients of tools.h.

Reply:
I have slightly modified the DoConcurrentLabelEnforce class (https://github.com/flang-compiler/f18/blob/master/lib/semantics/check-do.cc#L350) in check-do.cc and renamed it as LabelEnforce and moved to tools.h. The LabelEnforce class now has two customers do-concurrent checks in check-do.cc and critical checks in check-coarray.cc. This can also be used for the change-team construct.

The class (given a set of labels) provides an error message if there is a branch to label which is not in the given set. I can add this as a comment. Is this OK?

Reply:
That's a great one-sentence description, and it's not something I would have understood immediately just from the name of the class.